vnext/Microsoft.ReactNative/ReactNativeAppBuilder.cpp

```diff
@@ -16,51 +39,138 @@ using namespace winrt::Microsoft::UI;
 ReactNativeAppBuilder::ReactNativeAppBuilder() {
-  m_host = winrt::make<winrt::ReactNative::implementation::ReactNativeHost>();
+  m_reactNativeWin32App = winrt::make<implementation::ReactNativeWin32App>();
 }
 
+ReactNativeAppBuilder::~ReactNativeAppBuilder() {}
+
 winrt::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::AddPackageProviders(
     winrt::Windows::Foundation::Collections::IVector<winrt::Microsoft::ReactNative::IReactPackageProvider> const
         &packageProviders) {
   for (auto const &provider : packageProviders) {
-    m_host.PackageProviders().Append(provider);
+    m_reactNativeWin32App.ReactNativeHost().PackageProviders().Append(provider);
   }
 
   return *this;
 }
 
 winrt::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::SetReactInstanceSettings(
     winrt::Microsoft::ReactNative::ReactInstanceSettings const &settings) {
-  m_host.InstanceSettings(settings);
+  m_reactNativeWin32App.ReactNativeHost().InstanceSettings(settings);
 
   return *this;
 }
 
 winrt::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::SetCompositor(
     winrt::Microsoft::UI::Composition::Compositor const &compositor) {
-  m_compositor = compositor;
-
+  m_reactNativeWin32App.as<implementation::ReactNativeWin32App>().get()->Compositor(compositor);
   return *this;
 }
 
 winrt::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::SetAppWindow(
     winrt::Microsoft::UI::Windowing::AppWindow const &appWindow) {
-  m_appWindow = appWindow;
+  m_reactNativeWin32App.as<implementation::ReactNativeWin32App>().get()->AppWindow(appWindow);
+
+  return *this;
+}
+
+winrt::Microsoft::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::SetReactViewOptions(
+    winrt::Microsoft::ReactNative::ReactViewOptions const &reactViewOptions) {
+  m_reactViewOptions = reactViewOptions;
 
   return *this;
 }
 
-winrt::ReactNative::ReactNativeHost ReactNativeAppBuilder::Start() {
-  auto hwnd{winrt::UI::GetWindowFromWindowId(m_appWindow.Id())};
+winrt::ReactNative::ReactNativeWin32App ReactNativeAppBuilder::Build() {
+  if (m_reactNativeWin32App.Compositor() == nullptr) {
+    // Create a DispatcherQueue for this thread.  This is needed for Composition, Content, and
+    // Input APIs.
+    auto dispatcherQueueController =
+        winrt::Microsoft::UI::Dispatching::DispatcherQueueController::CreateOnCurrentThread();
+
+    m_reactNativeWin32App.as<implementation::ReactNativeWin32App>().get()->DispatchQueueController(
+        dispatcherQueueController);
+
+    // Create the compositor on behalf of the App Developer
+    auto compositor = winrt::Microsoft::UI::Composition::Compositor();
+    m_reactNativeWin32App.as<implementation::ReactNativeWin32App>().get()->Compositor(compositor);
+  }
+
+  // Create the AppWindow if the developer doesn't provide one
+  if (m_reactNativeWin32App.AppWindow() == nullptr) {
+    auto appWindow = winrt::Microsoft::UI::Windowing::AppWindow::Create();
+    appWindow.Title(L"SampleFabricApplication");
```

Comment:
Pretty sure most apps wont want this to be their title.

Reply:
Any suggestions for a default name? And yeah the title can updated anytime before the ReactNativeWin32::Start() is triggered, so the developers can change it later on.